src/QuadForm/Quad/ZGenusRep.jl

```diff
@@ -0,0 +1,1030 @@
+  for x in Iterators.drop(A, 1)
+    if all(y -> !is_isometric(x, y), A[1:count])
+      # In that case, x represents a new isometry class, so we stack it
+      # after the first isometry classes kept at the beginning of `A`
+      A[it[1]] = x
+      count += 1
+      it = iterate(idxs, it[2])
+    end
+  end
+  # Now the lattices to keep are at the first `count` entries of `A`
+  resize!(A, count)::typeof(A)
+end
+
+@doc raw"""
+    default_invariant_function(L::ZZLat) -> Tuple
+
+Return a list of isometry invariants of the definite lattice $L$. For now,
+the invariants by default are:
+- the (absolute) minimum of $L$;
+- the combinatorial data of the root sublattice of $L$;
+- the kissing numbe of $L$;
+- the order of the isometry group of $L$.
+"""
+function default_invariant_function(L::ZZLat)
+  m = minimum(L)
+  rlr, _ = root_lattice_recognition(L)
+  kn = kissing_number(L)::Int
+  ago = automorphism_group_order(L)::ZZRingElem
+  return (m, rlr, kn, ago)
+end
+
+###############################################################################
+#
+#  User interface
+#
+###############################################################################
+
+@doc raw"""
+    is_maximal_integral_bilinear(G::ZZGenus, p::ZZRingElem) -> Bool
+
+Given a genus `G` of integral $\mathbb{Z}$-lattices and a prime number
+`p`, return whether the lattice $L_p$ is maximal integral in
+$\mathbb{Q}L_p$, for the bilinear form on $L_p$.
+
+Note that this does not depend on a choice of a lattice $L$ in $G$,
+and this property can be directly check from the genus symbol.
+"""
+function is_maximal_integral_bilinear(G::ZZGenus, p::ZZRingElem)
+  @req is_integral(G) "Lattices in the genus must be globally integral"
+  sym = symbol(local_symbol(G, p))
+
+  # If the local lattice is unimodular, then the bilinear form on the
+  # discriminant group `q_p` is trivial and it has no isotropic vector.
+  #
+  # If the p-adic level `l = p^k` is greater or equal than p^2, then the
+  # bilinear form on `p^(k-1) * q_p` is totally isotropic, and `q_p` has an
+  # isotropic vector.
+  if sym[end][1] == 0
+    return true
+  elseif sym[end][1] >= 2
+    return false
+  end
+
+  # If the Jordan constituent of scale p has rank 1 then there is
+  # no isotropic vector for the bilinear form on the discriminant group.
+  #
+  # If it has rank 3 then as a consequence of Chevalley's theorem
+  # the bilinear form on the discriminant group has an isotropic vector.
+  if sym[end][2] == 1
+    return true
+  elseif sym[end][2] >= 3
+    return false
+  end
+
+  # For the rank 2, there are only two possible normal forms for the
+  # bilinear form on the discriminant group.
+  # In the case p=2, they both admit an isotropic vector
+  p == 2 && return false
+
+  # For the odd case, one of them has an isotropic vector but
+  # not the other one
+  sym[end][3] == jacobi_symbol(-1, p) && return false
+  return true
+end
+
+@doc raw"""
+    _is_isotropic(G::ZZGenus, p::ZZRingElem) -> Bool
```

Comment:
We have this functionality already.

Reply:
As far as I know, we have it for quadratic spaces, but I want to avoid constructing a space where I could only check it from the symbol. So it is the same code at the end.

Reply:
we have `is_isotropic(rational_isometry_class(::ZZGenus))`
please use that. no space is created